test/OpenTelemetry.Exporter.Jaeger.Tests/Implementation/JaegerUdpBatcherTests.cs

```diff
@@ -177,23 +194,38 @@ internal static JaegerSpan CreateTestPayloadJaegerSpan()
 
             var linkedSpanId = ActivitySpanId.CreateFromString("888915b6286b9c41".AsSpan());
 
-            var link = new Link(new SpanContext(
+            var activitySource = new ActivitySource(nameof(CreateTestPayloadJaegerSpan));
+
+            var tags = attributes.Select(kvp => new KeyValuePair<string, string>(kvp.Key, kvp.Value.ToString()));
+
+            var links = new[]
+            {
+                new ActivityLink(new ActivityContext(
                     traceId,
                     linkedSpanId,
-                    ActivityTraceFlags.Recorded));
+                    ActivityTraceFlags.Recorded)),
+            };
 
-            return new SpanData(
+            Activity activity = activitySource.StartActivity(
                 "Name",
-                new SpanContext(traceId, spanId, ActivityTraceFlags.Recorded),
-                parentSpanId,
-                SpanKind.Client,
-                startTimestamp,
-                attributes,
-                events,
-                new[] { link, },
-                null,
-                Status.Ok,
-                endTimestamp).ToJaegerSpan();
+                ActivityKind.Client,
+                parentContext: default,
+                tags,
+                links,
+                startTime: startTimestamp);
+
+            typeof(Activity).GetField("_traceId", BindingFlags.Instance | BindingFlags.NonPublic).SetValue(activity, traceId.ToHexString());
```

Comment:
@tarekgh @cijothomas @reyang Just following up on [this comment](https://github.com/dotnet/runtime/pull/39087#discussion_r455317748).

Here's the unit test code I was working on.

* Modifying `Links` is really hard after construction because it uses an internal type. I ended up going through `ActivitySource.StartActivity` just so I could pass them in. Also allowed me to specify ActivityKind.

* I needed this test to be deterministic so I set traceId, spanId, & parentId through reflection. I don't think there is a way in the API to set them otherwise.

Providing just as FYI. This isn't real-world usage obviously.

Reply:
`parentId` can be set using `activity.SetParentId("00-0123456789abcdef0123456789abcdef-0123456789abcdef-01");`

Reply:
Why not pass parentid when calling activitySource.StartActivity?

Reply:
`traceId` can be _passed in_ via `ActivityContext`.

Reply:
@cijothomas I don't know a public way of specifying `spanId` for an activity, and I believe it was intended.

Reply:
Thats right. No way to set `spanId`. `TraceId` and `ParentSpanId`, can be set while starting.

Reply:
@CodeBlanch @reyang @cijothomas , i tried to change to use the method SetParentId. When i do that, it seens it changes internally something that makes the test non-deterministic -> so, the bytearray gets different.

The code that i tried:
`activity.SetParentId(parentSpanId.ToHexString());`
`activity.SetParentId(traceId, spanId, ActivityTraceFlags.None);`
`activity.SetParentId(traceId, spanId, ActivityTraceFlags.Recorded);`

Reply:
@eddynaka You still need to use reflection to set spanid.

Reply:
@cijothomas i didn't remove the spanId reflection. I just removed the parentId reflection and tried to use the above calls.

Reply:
@eddynaka confirmed reflection is working to make this predictable. We can follow up separately to see why setting parentid using the Setparentid is not working.